sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs

```diff
@@ -140,7 +140,7 @@ public BinaryData Body
             get => AmqpMessage.GetBody();
             set
             {
-                AmqpMessage.Body = new AmqpMessageBody(new ReadOnlyMemory<byte>[] { value });
+                AmqpMessage.Body = new AmqpMessageBody(new BodyMemory(new ReadOnlyMemory<byte>[] { value }));
```

Comment:
The direct body modifications from outside are not yet covered. I wanted to first see what you think about this